src/corona_hakab_model/agent.py

```diff
@@ -81,17 +81,20 @@ def remove_agent(self, agent):
 
 
 class TrackingCircle(Circle):
-    __slots__ = ("agents",)
+    __slots__ = ("agents", "ever_visited")
 
     def __init__(self):
         super().__init__()
         self.agents = set()
+        # done to count how many different agents ever visited a given state
+        self.ever_visited = set()
 
     def add_agent(self, agent):
         super().add_agent(agent)
         if agent in self.agents:
             raise ValueError("DuplicateAgent")
         self.agents.add(agent)
+        self.ever_visited.add(agent.index)
```

Comment:
why store indices instead of agents?

Reply:
because we only care about the len of it, and ints take less memory than references in python (i think)

Reply:
all values in python require the same length. an "agent" is just an address pointer

Reply:
changed to store agents